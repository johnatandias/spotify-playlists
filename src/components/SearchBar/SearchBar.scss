@import '../../variables';

.searchBar {
  position: relative;
  display: flex;
  width: 100%;
  height: 60px;
  min-height: 60px;
  background-color: $search-bar-background-color;
  min-width: 622px;

  &::before {
    content: " ";
    position: absolute;
    left: 25px;
    top: 14px;
    width: 2px;
    height: 36px;
    transform: scale(0);
    z-index: 1;
    background: $highlight-color;
    transition: all 0.5s cubic-bezier(0.87, -0.41, 0.19, 1.44);
  }

  &__input {
    display: block;
    margin: 0;
    color: $primary-text-color;
    background: transparent;
    font-family: inherit;
    border: none;
    padding: 15px;
    font-size: 30px;
    font-weight: 500;
    caret-color: #1ab26b;
    width: 100%;
    height: 100%;

    &::placeholder {
      color: #5a5a5a;
    }

    &:focus {
      outline: none;
    }
  }
}
