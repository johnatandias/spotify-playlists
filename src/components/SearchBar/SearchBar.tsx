import React from 'react';
import './SearchBar.scss';

interface SearchBar {
  placeholder: string,
  onSearch: Function
}

export const SearchBar = ({ placeholder, onSearch }: SearchBar) => (
  <div className="searchBar">
    <input
      type="text"
      placeholder={placeholder.toUpperCase()}
      onChange={(evt) => onSearch(evt.target.value)}
      className="searchBar__input"
    />
  </div>
);
