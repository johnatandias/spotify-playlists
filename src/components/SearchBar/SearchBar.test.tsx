import React from "react";
import { shallow } from 'enzyme';
import { SearchBar } from './SearchBar';

describe('SearchBar component', () => {
  it('Renders without crashing', () => {
    const wrapper = shallow(<SearchBar placeholder="Pesquisar..." onSearch={() => { }} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should handle the input event', () => {
    window.alert = jest.fn();
    const callAlert = (value: string) => alert(value);
    const wrapper = shallow(<SearchBar placeholder="Pesquisar..." onSearch={callAlert} />);
    const value = 'Changed';
    wrapper.find('input').simulate('change', { target: { value } });
    expect(window.alert).toHaveBeenCalledWith(value);
  });
});