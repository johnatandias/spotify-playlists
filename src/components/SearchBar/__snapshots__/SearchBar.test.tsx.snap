// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component Renders without crashing 1`] = `ShallowWrapper {}`;
